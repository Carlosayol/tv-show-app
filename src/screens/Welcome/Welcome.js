import { View, Text, StyleSheet, Modal, KeyboardAvoidingView } from "react-native";
import React, { useState } from "react";
import { MaterialIcons } from "@expo/vector-icons";
import { useNavigation } from "@react-navigation/native";

import CustomButton from "../../components/CustomButton.js";
import CustomInput from "../../components/CustomInput.js";
import Constants from "../../utils/Constants";

const validCredentials = [
  {
    user: "maria",
    password: "password",
  },
  {
    user: "pedro",
    password: "123456",
  },
];

const Welcome = () => {
  const [modalOpen, setModalOpen] = useState(false);
  const [user, setUser] = useState("");
  const [password, setPassword] = useState("");

  const navigation = useNavigation();

  const onSignUpPressed = () => {
    console.log("Sign up");
  };

  const onLoginPressed = () => {
    setModalOpen(false);
    navigation.navigate("Home");

    // if (validCredentials.some((cred) => cred.user === user && cred.password === password)) {
    //   setModalOpen(false);
    //   navigation.navigate("Home");
    // } else {
    //   console.warn("Wrong credentials");
    // }
  };

  const onForgotPressed = () => {
    console.log("forgot");
  };

  return (
    <View style={styles.root}>
      <Text style={{ fontFamily: "GilroyBold", fontSize: 24, height: 100 }}>Welcome!</Text>
      <CustomButton onPress={onSignUpPressed} text="Sign up" type="Primary" />
      <CustomButton onPress={() => setModalOpen(true)} text="Log in" type="Secondary" />
      <CustomButton onPress={onForgotPressed} text="Forgot password?" type="Tertiary" />

      <Modal visible={modalOpen} animationType="slide" transparent={true}>
        <KeyboardAvoidingView behavior="padding" enabled>
          <View style={styles.modal}>
            <MaterialIcons style={styles.modalToggle} name="close" size={24} onPress={() => setModalOpen(false)} />
            <CustomInput placeholder="Name" value={user} setValue={setUser} />
            <CustomInput placeholder="Password" value={password} setValue={setPassword} secure />
            <CustomButton onPress={onLoginPressed} text="Log in" type="Secondary" />
          </View>
        </KeyboardAvoidingView>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: "#e8e8e8",
    justifyContent: "center",
    alignItems: "center",
    padding: 20,
  },
  modal: {
    padding: 20,
    alignItems: "center",
    marginTop: "100%",
    height: "100%",
    backgroundColor: Constants.secondaryColor,
    opacity: 0.8,
    borderRadius: 15,
  },
  modalToggle: {
    alignSelf: "flex-end",
    margin: 10,
    color: Constants.tertiaryColor,
  },
});

export default Welcome;
