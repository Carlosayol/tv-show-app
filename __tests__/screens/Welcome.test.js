import React from "react";
import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react-native";

import Welcome from "../../src/screens/Welcome";

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => ({ useNavigation: () => ({ navigate: mockedNavigate }) }));

describe("Welcome Screen", () => {
  afterEach(cleanup);

  it("render base element", () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Welcome />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("find login button", () => {
    const { getByText } = render(<Welcome />);
    expect(getByText("Log in")).toBeDefined();
  });
});
