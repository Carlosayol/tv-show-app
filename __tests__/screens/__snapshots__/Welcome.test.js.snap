// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "uri": "https://www.themoviedb.org/t/p/original/JFDrOxBFxeJrALLKWHWcM8i1sp.jpg",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0, 0.40)",
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "GilroyBold",
            "fontSize": 24,
            "height": 100,
          }
        }
      >
        Welcome!
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 25,
              "marginVertical": 15,
              "padding": 15,
              "width": 200,
            },
            {
              "backgroundColor": "#FFD233",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Gilroy",
                "fontSize": 18,
              },
              undefined,
            ]
          }
        >
          Sign up
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 25,
              "marginVertical": 15,
              "padding": 15,
              "width": 200,
            },
            {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Gilroy",
                "fontSize": 18,
              },
              undefined,
            ]
          }
        >
          Log in
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 25,
              "marginVertical": 15,
              "padding": 15,
              "width": 200,
            },
            {
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Gilroy",
                "fontSize": 18,
              },
              {
                "color": "#8C8C8C",
                "fontFamily": "Gilroy",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Forgot password?
        </Text>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      />
    </View>
  </View>
</View>
`;
